lib/galaxy/tools/cache.py

```diff
@@ -147,7 +147,7 @@ def tool_shed_repositories(self):
         except AttributeError:
             self.rebuild()
             repositories = self.cache.repositories
-        if repositories and not repositories[0]._sa_instance_state._attached:
+        if repositories and any(_ for _ in repositories if hasattr(_, '_sa_instance_state') and not _._sa_instance_state.attached):
```

Comment:
```suggestion
        if repositories and any(_ for _ in repositories if hasattr(_, '_sa_instance_state') and not _._sa_instance_state._attached):
```

or we can use a more stable API like https://docs.sqlalchemy.org/en/13/orm/session_state_management.html#getting-the-current-state-of-an-object